src/psyclone/dynamo0p3.py

```diff
@@ -696,26 +688,28 @@ def __init__(self, arg_type):
                            arg_type))
             self._function_space2 = arg_type.args[3].name
             # Test allowed accesses for operators
-            if self._access_descriptor.name.lower() == "gh_inc":
+            if self._access_descriptor.name == AccessType.INC:
                 raise ParseError(
                     "In the dynamo0.3 API operators cannot have a 'gh_inc' "
                     "access because they behave as discontinuous quantities")
 
         # Scalars
-        elif self._type in VALID_SCALAR_NAMES:
+        elif self._type in GH_VALID_SCALAR_NAMES:
             if len(arg_type.args) != 2:
                 raise ParseError(
                     "In the dynamo0.3 API each meta_arg entry must have 2 "
                     "arguments if its first argument is gh_{{r,i}}scalar, but "
                     "found {0} in '{1}'".format(len(arg_type.args), arg_type))
             # Test allowed accesses for scalars (read_only or reduction)
-            if self._access_descriptor.name not in ["gh_read"] + \
-               VALID_REDUCTION_NAMES:
+            if self._access_descriptor.name not in [AccessType.READ] + \
+               AccessType.get_valid_reduction_modes():
+                api_name = rev_access_mapping[self._access_descriptor.name]
```

Comment:
I really prefer to have 'api' in the name, since access_name() does not indicate that AccessType.INC.access_name() can be different depending on the API loaded. What about:
api_specific_name()  ?

Reply:
How about `api_access_name`?

Reply:
While I have to admit to prefer api_specific_name (the typical usage of this functions is:
```
  access.api_specific_name()   # vs:
  access.api_access_name()
```
But it doesn't really mattter, so api_access_name() it is now.

Reply:
If I understood you correct, you actually agreed to api_specific_name for the function, and then renaming the local variables. I've done that.

Reply:
Yes, I did - thanks!